src/posts/blog/3-hepa-filters/HepaCalculator.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react'
+import { Col, Row } from 'react-bootstrap'
+
+interface Dimensions {
+  length: number
+  width: number
+  height: number
+  unit: string
+}
+
+const DimensionControl: React.FunctionComponent<{
+  label: string
+  value: number
+  setter: (val: number) => void
+  unit?: string
+}> = ({ label, value, setter, unit = 'ft' }) => {
+  return (
+    <>
+      <label>
+        {label}({unit}):
+        <input
+          className="form-control form-control-lg col-md-6"
+          type="number"
+          value={value}
+          onChange={(e) => setter(Math.max(0, parseFloat(e.target.value)))}
+        />
+      </label>
+    </>
+  )
+}
+
+const FEET_PER_METER = 3.28084
+const CFM_PER_CMH = FEET_PER_METER ** 3 / 60
+
+export const HepaCalculator: React.FunctionComponent = () => {
+  const [dimensions, setDimensions] = useState<Dimensions>({
+    length: 12,
+    width: 10,
+    height: 8,
+    unit: 'ft',
+  })
+
+  const toggleUnits = () => {
+    let unit
+    let multiplier
+    if (dimensions.unit === 'ft') {
+      unit = 'm'
+      multiplier = 1 / FEET_PER_METER
+    } else {
+      unit = 'ft'
+      multiplier = FEET_PER_METER
+    }
+    setDimensions({
+      unit,
+      // Round to one decimal place
+      length: Math.round(dimensions.length * multiplier * 10) / 10,
+      width: Math.round(dimensions.width * multiplier * 10) / 10,
+      height: Math.round(dimensions.height * multiplier * 10) / 10,
+    })
+  }
+
+  let cfm // cubic feet per minute
+  let cmh // cubic feet per hour (yes, metric folks use a different time base)
+  if (dimensions.unit === 'ft') {
+    cfm = (5 * (dimensions.length * dimensions.width * dimensions.height)) / 60
+    cmh = cfm / CFM_PER_CMH
+  } else {
+    cmh = 5 * (dimensions.length * dimensions.width * dimensions.height)
+    cfm = cmh * CFM_PER_CMH
+  }
+
+  cfm = cfm.toFixed(0)
+  cmh = cmh.toFixed(0)
+
+  return (
+    <div className="card">
+      <Row>
+        <Col>
+          <DimensionControl
+            label="Width"
+            value={dimensions.width}
+            unit={dimensions.unit}
+            setter={(val: number) =>
+              setDimensions({ ...dimensions, width: val })
+            }
+          />
+        </Col>
+        <Col>
+          <DimensionControl
+            label="Length"
+            value={dimensions.length}
+            unit={dimensions.unit}
+            setter={(val: number) =>
+              setDimensions({ ...dimensions, length: val })
+            }
+          />
+        </Col>
+        <Col>
+          <DimensionControl
+            label="Height"
+            value={dimensions.height}
+            unit={dimensions.unit}
+            setter={(val: number) =>
+              setDimensions({ ...dimensions, height: val })
+            }
+          />
+        </Col>
+        <Col>
+          <strong>Minimum CADR</strong>:
```

Comment:
I wonder if we could make this more clear by not using an acronym? What do you think?

It makes it feel more intimidating. I know when I first open an article like this I skin it in 5-10 seconds and often play with something like a calculator first before even reading. It'd be interesting to me to have something clearer like: "if your room.is this big FIELDS then your air purifier needs to be rates at this level or better RESULT"

Reply:
Ah yeah, I could go either way "Minimum Flow Rate" is nice because it's human readable. CADR is nice because it's what I see on the actual product pages, so making it clear that it's what you should be looking for is helpful.

Minimum Flow Rate (CADR) is too long.

Reply:
(Changed to flow rate)